### PR TITLE
cli(engine): move master-client-cli to tiflow sub command

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: ref #6022

### What is changed and how it works?

- migrate df-master-client to tiflow cli

#### Tests 

 - Unit test
 - Manual test (use `tiflow cli job`)

```bash
:) apple@ubuntu-work in [~/work/code/tiflow/bin] on git:migrate-tiflow-cli-cmd o [12:18:40]
➜  ./tiflow cli job create --master-addrs 127.0.0.1:10341 --job-type FakeJob --job-config config/fake_job.json
[2022/07/05 12:18:42.192 +08:00] [INFO] [client.go:103] ["add new server master client"] [addr=127.0.0.1:10341]
[2022/07/05 12:18:42.198 +08:00] [WARN] [cli_job.go:81] ["tenant-id is empty, use default tenant id"]
[2022/07/05 12:18:42.198 +08:00] [WARN] [cli_job.go:87] ["project-id is empty, use default project id"]
[2022/07/05 12:18:42.202 +08:00] [INFO] [cli_job_create.go:101] ["create job successfully"] [resp="job_id:\"49f3ad08-aff2-4fe8-8527-a4ce50e23091\" "]

:) apple@ubuntu-work in [~/work/code/tiflow/bin] on git:migrate-tiflow-cli-cmd o [12:18:42]
➜  ./tiflow cli job query --master-addrs 127.0.0.1:10341 --job-id 49f3ad08-aff2-4fe8-8527-a4ce50e23091
[2022/07/05 12:18:51.840 +08:00] [INFO] [client.go:103] ["add new server master client"] [addr=127.0.0.1:10341]
[2022/07/05 12:18:51.841 +08:00] [WARN] [cli_job.go:81] ["tenant-id is empty, use default tenant id"]
[2022/07/05 12:18:51.841 +08:00] [WARN] [cli_job.go:87] ["project-id is empty, use default project id"]
[2022/07/05 12:18:51.842 +08:00] [INFO] [cli_job_query.go:64] ["query job successfully"] [resp="tp:5 config:\"{\\n    \\\"job-name\\\": \\\"test-fake-job-1\\\",\\n    \\\"worker-count\\\": 3,\\n    \\\"target-tick\\\": 20000\\n}\\n\" status:online job_master_info:<id:\"49f3ad08-aff2-4fe8-8527-a4ce50e23091\" executor_id:\"35cb25cc-4bb1-4f40-b556-9d6aa3f437bc\" status:\"{\\\"seq-id\\\":13,\\\"created-at\\\":\\\"2022-07-05T12:18:42.211+08:00\\\",\\\"updated-at\\\":\\\"2022-07-05T12:18:42.211+08:00\\\",\\\"project-id\\\":\\\"dfe_proj_default\\\",\\\"job-id\\\":\\\"dataflow-engine-job-manager\\\",\\\"id\\\":\\\"49f3ad08-aff2-4fe8-8527-a4ce50e23091\\\",\\\"type\\\":3,\\\"code\\\":1,\\\"error-message\\\":\\\"\\\",\\\"ext-bytes\\\":\\\"eyIwNDE2NzQ1Yy0xMThkLTQ4NGQtOWFmYy1mMzYzZDcyN2JkMDgiOnsiYnVzaW5lc3MtaWQiOjIsInRpY2siOjE2OSwiY2hlY2twb2ludCI6eyJ0aWNrIjowLCJyZXZpc2lvbiI6MCwibXZjYy1jb3VudCI6MCwidmFsdWUiOiIifX0sIjE1N2FkODM4LTU5ZmMtNGJiMi04N2IwLWJiMTQ0MDA4YmRjOCI6eyJidXNpbmVzcy1pZCI6MSwidGljayI6MTY4LCJjaGVja3BvaW50Ijp7InRpY2siOjAsInJldmlzaW9uIjowLCJtdmNjLWNvdW50IjowLCJ2YWx1ZSI6IiJ9fSwiZGRjMjVlZDQtMzJmZS00YjE1LWI5Y2EtMDQyYTg1MmFhNzU3Ijp7ImJ1c2luZXNzLWlkIjowLCJ0aWNrIjoxNjksImNoZWNrcG9pbnQiOnsidGljayI6MCwicmV2aXNpb24iOjAsIm12Y2MtY291bnQiOjAsInZhbHVlIjoiIn19fQ==\\\"}\" > "]

:) apple@ubuntu-work in [~/work/code/tiflow/bin] on git:migrate-tiflow-cli-cmd o [12:18:51]
➜  ./tiflow cli job pause --master-addrs 127.0.0.1:10341 --job-id 49f3ad08-aff2-4fe8-8527-a4ce50e23091
[2022/07/05 12:18:58.414 +08:00] [INFO] [client.go:103] ["add new server master client"] [addr=127.0.0.1:10341]
[2022/07/05 12:18:58.414 +08:00] [WARN] [cli_job.go:81] ["tenant-id is empty, use default tenant id"]
[2022/07/05 12:18:58.414 +08:00] [WARN] [cli_job.go:87] ["project-id is empty, use default project id"]
[2022/07/05 12:18:58.415 +08:00] [INFO] [cli_job_pause.go:64] ["pause job request is sent"] [resp=]

:) apple@ubuntu-work in [~/work/code/tiflow/bin] on git:migrate-tiflow-cli-cmd o [12:18:58]
➜  ./tiflow cli job query --master-addrs 127.0.0.1:10341 --job-id 49f3ad08-aff2-4fe8-8527-a4ce50e23091
[2022/07/05 12:19:02.525 +08:00] [INFO] [client.go:103] ["add new server master client"] [addr=127.0.0.1:10341]
[2022/07/05 12:19:02.526 +08:00] [WARN] [cli_job.go:81] ["tenant-id is empty, use default tenant id"]
[2022/07/05 12:19:02.526 +08:00] [WARN] [cli_job.go:87] ["project-id is empty, use default project id"]
[2022/07/05 12:19:02.527 +08:00] [INFO] [cli_job_query.go:64] ["query job successfully"] [resp="tp:5 config:\"{\\n    \\\"job-name\\\": \\\"test-fake-job-1\\\",\\n    \\\"worker-count\\\": 3,\\n    \\\"target-tick\\\": 20000\\n}\\n\" status:stopped "]
```

### Release note 

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 .PHONY: build test check clean fmt cdc kafka_consumer coverage \
 	integration_test_build integration_test integration_test_mysql integration_test_kafka bank \
 	dm dm-master dm-worker dmctl dm-syncer dm_coverage \
-	engine tiflow df-master-client df-demo df-chaos-case
+	engine tiflow df-demo df-chaos-case
```
Comment (lonng) [inline Makefile:5]: Can we rename the `df-demo df-chaos-case` to `tiflow-demo tiflow-chaos-case`?

Comment (amyangfei) [reply Makefile:5]: Yes, they will be renamed in another PR, in case of this PR too large.

File: engine/pkg/cmd/cli/cli_job_create.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	"github.com/pingcap/tiflow/engine/enginepb"
+	engineModel "github.com/pingcap/tiflow/engine/model"
+	cmdcontext "github.com/pingcap/tiflow/pkg/cmd/context"
+	"github.com/pingcap/tiflow/pkg/errors"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+)
+
+// createJobOptions defines flags for job create.
+type createJobOptions struct {
```
Comment (CharlesCheung96) [inline engine/pkg/cmd/cli/cli_job_create.go:31]: `createJobOptions` and `jobGeneralOptions` should be named in the same style, maybe `jobCreateOptions ` is appropriate?

**Comment by CharlesCheung96** [comment]
image
The help message should be prompted if no parameters are entered.

File: engine/pkg/cmd/cli/cli_job.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+	"github.com/pingcap/tiflow/engine/client"
+	"github.com/pingcap/tiflow/engine/pkg/tenant"
+	"github.com/pingcap/tiflow/pkg/errors"
+)
+
+// jobGeneralOptions defines some general options of job management
+type jobGeneralOptions struct {
+	projectID string
+	tenantID  string
```
Comment (CharlesCheung96) [inline engine/pkg/cmd/cli/cli_job.go:31]: It's redundant. The `ProjectInfo` already contains `tenantID` field.

Comment (amyangfei) [reply engine/pkg/cmd/cli/cli_job.go:31]: They are different, `tenantID` comes from command line parameter, `tenant` is generated from `projectID` and `tenantID`

**Comment by amyangfei** [comment]
PTAL @CharlesCheung96 @maxshuang

File: engine/pkg/cmd/executor/executor.go [hunk 4]
```diff
@@ -116,6 +116,8 @@ func (o *options) complete(cmd *cobra.Command) error {
 
 	cmd.Flags().Visit(func(flag *pflag.Flag) {
 		switch flag.Name {
+		case "name":
```
Comment (maxshuang) [inline engine/pkg/cmd/executor/executor.go:119]: What's the difference of 'Name' and 'uuid' for executor?

**Review by maxshuang — APPROVED** [review]

File: engine/pkg/cmd/executor/executor.go (hunk 4 above)
Comment (amyangfei) [reply engine/pkg/cmd/executor/executor.go:119]: It seems redundant. Maybe we can remove one of them.
This PR doesn't change any logic, will leave a TODO comment.

File: engine/pkg/cmd/cli/cli_job.go (hunk 3 above)
Comment (maxshuang) [reply engine/pkg/cmd/cli/cli_job.go:31]: @CharlesCheung96 We need 'tenantID' and 'projectID' for Billing system in the future. They have different meaning actually.

In our implement, sometimes we only use 'projectID' for a unique identifier for multi-project isolation.

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**